hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -88,6 +96,21 @@ private AbfsClient(final URL baseUrl, final SharedKeyCredentials sharedKeyCreden
     this.accountName = abfsConfiguration.getAccountName().substring(0, abfsConfiguration.getAccountName().indexOf(AbfsHttpConstants.DOT));
     this.authType = abfsConfiguration.getAuthType(accountName);
 
+    String encryptionKey = this.abfsConfiguration.getClientProvidedEncryptionKey();
+    if (encryptionKey != null) {
+      this.clientProvidedEncryptionKey = encryptionKey;
+      try {
+        MessageDigest digester = MessageDigest.getInstance("SHA-256");
+        digester.update(encryptionKey.getBytes());
+        this.clientProvidedEncryptionKeySHA = Base64.getEncoder().encodeToString(digester.digest());
+      } catch (NoSuchAlgorithmException e) {
+        throw new IOException(e);
```

Comment:
* will this include information about the requested algorithm? 
* will this surface when the JVM doesn' t have the unrestricted crypto extension settings? If so, error text should mention it.

Reply:
Looks like this is a common Exception, do we need to mention explicitly?
https://stackoverflow.com/questions/48040299/java-security-nosuchalgorithmexception-sha3-384-messagedigest-not-available-w